bubble_main.rb

```diff
@@ -0,0 +1,46 @@
+test = [1, 100, 45, 4, 57, 0, 6, 6]
+
+def bubble_sort(arr)
+  puts 'Bubble Sort'
+  swap = true
+  while swap
+    swap = false
+    (arr.length - 1).times do |i|
+      if arr[i] > arr[i + 1]
+        swap = true
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      end
+    end
+  end
+
+  arr
+end
+
+int_arr = bubble_sort(test)
+print 'Sorted Array -> '
+puts int_arr.inspect
+print "\n\n"
+
+def bubble_sort_by(arr)
+  puts 'Bubble Sort By'
+  swap = true
+  while swap
+    swap = false
+    (arr.length - 1).times do |i|
+      compare = yield(arr[i], arr[i + 1])
+      # puts compare
+      if compare == 1
+        swap = true
+        arr[i], arr[i + 1] = arr[i + 1], arr[i]
+      end
+    end
+  end
+  arr
+end
```

Comment:
There is still a problem with this method kindly check below


`
       expected: ["hi", "hey", "hello"]
            got: ["hi", "hello", "hey"]
     `
### hint: kindly double check the implementation in your block